plugin/clipboard-filter/src/main/java/org/fcitx/fcitx5/android/plugin/clipboard_filter/ClearURLs.kt

```diff
@@ -5,25 +5,30 @@
 package org.fcitx.fcitx5.android.plugin.clipboard_filter
 
 import android.content.res.AssetManager
+import android.net.Uri
+import android.net.UrlQuerySanitizer
 import android.util.Log
 import kotlinx.serialization.KSerializer
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.builtins.MapSerializer
 import kotlinx.serialization.json.Json
 import kotlinx.serialization.serializer
 import java.net.URLDecoder
```

Comment:
I'm afraid it will meet the same issue as https://github.com/lz233/Tarnhelm/issues/46. Unlike JavaScript, Java's URLEncoder/URLDecoder use application/x-www-form-urlencoded rather than RFC 3986. Use encode()/decode() of java.net.URI instead.

Reply:
I had a feeling that it would be problematic when writing this. Fine. Let's use ktor https://api.ktor.io/ktor-http/io.ktor.http/encode-u-r-l-query-component.html

Reply:
Forget ktor, the `buildUrl` API design is so bad. I'd just use `android.net.Uri.{encode,decode}`